### PR TITLE
Add builds-v2 API

### PR DESCRIPTION
**Pull request by farhan-sauce**
### Changes

- Add builds-v2 API
- Add team-management API
- Remove builds-v1 API as it is deprecated
- Support existing commands to avoid breaking changes

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -128,83 +147,38 @@ export default class SauceLabs {
...
-                const optionName = camelCase(optionParam.name)
-                const option = options[optionName]
-                const isRequired = Boolean(optionParam.required) || (typeof optionParam.required === 'undefined' && typeof optionParam.default === 'undefined')
-                if ((isRequired || option) && !isValidType(option, expectedType)) {
-                    throw new Error(`Expected parameter for option '${optionName}' from type '${expectedType}', found '${typeof option}'`)
-                }
+    async _listBuilds(username, args) {
+        const { id: userId } = await this._getUserByUsername({ username })
+        const params = { userId, ...args }
+        const { builds } = await this._callAPI('getBuildsV2', 'vdc', params)
```
Comment (mciastek) [inline src/index.js:156]: I cannot find `getBuildsV2` method in files. Will it work without method definition?

File: src/index.js [hunk 2]
```diff
@@ -128,83 +147,38 @@ export default class SauceLabs {
...
+    }
 
-                if (typeof option !== 'undefined') {
-                    bodyMap.set(optionParam.name, option)
-                }
-            }
+    async _listBuildFailedJobs(username, buildId, args) {
+        const { id: userId } = await this._getUserByUsername({ username })
+        const params = { userId, faulty: true, ...args };
+        const { jobs: buildJobs } = await this._callAPI('getBuildsJobsV2', 'vdc', buildId, params)
```
Comment (mciastek) [inline src/index.js:163]: I think `getBuildsJobsV2` is missing too, right?

File: src/index.js [hunk 3]
```diff
@@ -128,83 +147,38 @@ export default class SauceLabs {
...
-                    ),
-                    responseType: 'json'
-                })
-                return response.body
-            } catch (err) {
-                throw new Error(`Failed calling ${propName}: ${err.message}, ${err.response && err.response.body}`)
-            }
+    async _getUserByUsername({ username }) {
+        try {
+            const { results: [user] } = await this._callAPI('getUsersV1', { username })
```
Comment (mciastek) [inline src/index.js:178]: `getUsersV1` is also missing. Is it defined in some external package?

File: apis/teamManagement.json [hunk 4]
```diff
@@ -0,0 +1,339 @@
...
+        },
+        "summary":"Get List of Teams",
+        "tags":[
+          "Team-Management"
+        ]
+      }
+    },
+    "/users":{
+      "get":{
+        "operationId":"get_users_v1",
```
Comment (farhan-sauce) [inline apis/teamManagement.json:282]: @mciastek this operationId can be used as `getUsersV1`.

File: apis/builds.json [hunk 5]
```diff
@@ -0,0 +1,610 @@
...
+      "in":"query",
+      "name":"faulty",
+      "type":"boolean",
+      "required": false
+    }
+  },
+  "paths": {
+    "/{build_source}/":{
+      "get":{
+        "operationId":"get_builds_v2",
```
Comment (farhan-sauce) [inline apis/builds.json:352]: @mciastek  This will be used as `getBuildsV2`.

File: apis/builds.json [hunk 6]
```diff
@@ -0,0 +1,610 @@
...
+        },
+        "summary":"Get Build Jobs",
+        "tags":[
+          "Build"
+        ]
+      }
+    },
+    "/{build_source}/{build_id}/jobs/":{
+      "get":{
+        "operationId":"get_builds_jobs_v2",
```
Comment (farhan-sauce) [inline apis/builds.json:523]: same for this as well, `get_builds_jobs_v2` operationId can be used as `getBuildsJobsV2`.

Comment (mciastek) [reply apis/builds.json:523]: Thanks for explanation 🙇

**Review by mciastek — APPROVED** [review]

**Comment by farhan-sauce** [comment]
I have used  https://github.com/saucelabs/node-saucelabs/pull/112 as reference.

**Pull request merged**